packages/events/defaultSubscribers/dashboard.js

```diff
@@ -0,0 +1,44 @@
+const ora = require("ora");
+
+module.exports = {
+  initialization: function () {
+    this.logger = (...args) => {
+      if (this.quiet) return;
+      console.log(...args);
+    };
+
+    this.pendingTransactions = {};
+  },
+  handlers: {
+    "rpc:request": [
+      function (payload) {
+        if (payload.method === "eth_sendTransaction") {
+          // TODO: Do we care about ID collisions?
+          this.pendingTransactions[payload.id] = payload;
+
+          this.spinner = ora({
+            text: `Waiting for transaction signature. Please check your wallet for a transaction approval message.`,
+            color: "red"
+          });
+        }
+      }
+    ],
+    "rpc:result": [
+      function (payload, error, result) {
```

Comment:
Rather than passing positional arguments, could we do this as type

```typescript
function (options:
  & { payload: any; }
  & (
    | { error: Error; /* (is this right?) */ }
    | { result: any }
  )
)
```

(I know this is JS, but my point is really just... can we used named options instead of positionals?)

Reply:
I mainly just did this because I was following the signature of the `preHook` and `postHook` functions in the provider wrapper, but yeah - I like that better.

Reply:
also as an aside, the `error` is always `TruffleError` there.

Reply:
Fortunately we don't have to worry about the types for real yet :)